plugins/modules/apt_rpm.py

```diff
@@ -179,8 +196,11 @@ def query_package_provides(module, name):
         name = local_rpm_package_name(name)
 
     rc, out, err = module.run_command("%s -q --provides %s" % (RPM_PATH, name))
-    return rc == 0
-
+    if rc == 0:
+        if check_package_version(module, name):
+            return True
+    else:
+        return False
```

Comment:
Right now if `rc == 0` but your function returns a falsish value, this function returns `None`. Better change this to
```suggestion
    return False
```
to return `False` instead.